aodntools/timeseries_products/hourly-timeseries.py

```diff
@@ -0,0 +1,554 @@
+    nc_aggregated = xr.merge([nc_metadata, nc_data])
+
+    ## add global attributes
+    add_attribute = {'rejected_files': "\n".join(rejected_files)}
+    nc_aggregated.attrs = set_globalattr(nc_aggregated, globalattr_file, site_code, add_attribute, parameter_names)
+
+
+    ## add variable attributes
+    add_variable_attribute = {'PRES_REL': {'applied_offset_by_instrument': applied_offset}}
+    parameter_names_all = list(set(parameter_names_all))
+    variable_attributes = set_variableattr(parameter_names_all + ['TIME'], variable_attribute_dictionary, add_variable_attribute)
+
+    time_units = variable_attributes['TIME'].pop('units')
+    time_calendar = variable_attributes['TIME'].pop('calendar')
+
+    for variable in parameter_names_all:
+        ancillary_variables_attr = []
+        nc_aggregated[variable].attrs = variable_attributes[variable]
+        nc_aggregated[variable].attrs['long_name'] = function_dict[variable] + " " + nc_aggregated[variable].attrs['long_name']
+        nc_aggregated[variable].attrs.update({'cell_methods': 'TIME:' + function_dict[variable] + ' (interval: 1 hr comment: time mid point)'})
+        for stat_method in function_stats:
+            variable_stat_name = variable + "_" + stat_method
+            ancillary_variables_attr += [variable_stat_name]
+            nc_aggregated[variable_stat_name].attrs = nc_aggregated[variable].attrs
+            nc_aggregated[variable_stat_name].attrs['long_name'] = stat_method + ' data value in the bin, after rejection of flagged data'
+            nc_aggregated[variable_stat_name].attrs['cell_methods'] = 'TIME:' +  stat_method
+        nc_aggregated[variable].attrs.update({'ancillary_variables': " ".join(ancillary_variables_attr)})
+
+    ## create the output file name and write the aggregated product as netCDF
+    facility_code = get_facility_code(files_to_aggregate[0])
+    data_code = "".join(sorted(list(set(data_codes))))
+    product_type = 'hourly-timeseries'
+    file_version = 2
+    ncout_filename = generate_netcdf_output_filename(nc=nc_aggregated, facility_code=facility_code, data_code=data_code,
+                                                     VoI="all-variables", site_code=site_code,
+                                                     product_type=product_type, file_version=file_version)
+    ncout_path = os.path.join(file_path, ncout_filename)
+
+    encoding = {'TIME': {'_FillValue': False,
+                         'units': time_units,
+                         'calendar': time_calendar},
+                'LONGITUDE': {'_FillValue': False},
+                'LATITUDE': {'_FillValue': False},
+                'instrument_id': {'dtype': '|S256'},
+                'source_file': {'dtype': '|S256'}}
+
+    write_netCDF_aggfile(nc_aggregated, ncout_path, encoding, file_path)
+
+
+
+    return ncout_path, bad_files
+
+
+if __name__ == "__main__":
+
+
+    site_code = 'GBRPPS'
+    #fnames = 'files_local5.txt'
+    fnames = 'files_local5.txt'
```

Comment:
Please remove hard-coded input parameters.

Reply:
removed

Reply:
changed to command line arg parser